pkg/cdi/annotations.go

```diff
@@ -17,9 +17,9 @@
 package cdi
 
 import (
+	"errors"
```

Comment:
Should we use `fmt.Errorf` over `errors.New` for all errors even if no formatting is performed?

Reply:
Would there be some real benefit to it ? This `errors` is golang stdlib (so always present/available by the toolchain) and we anyway depend on it because we use `errors.Is()` elsewhere in the code.

Reply:
Was more of a general question. Not a blocker.

Reply:
a related comment is to make the cdi errors hierarchy so that, e.g., `goerr113` linter is happy. In particular,
"`error` should not be created dynamically from scratch but by the wrapping the static (package-level) error."

```
$ golangci-lint run --disable-all -E goerr113
```

Reply:
@mythi I may have been a bit hasty in resolving this thread. I have created #102 to address the lint errors. I think that this is out of scope of this PR and also doesn't make the situation worse than what it currently is.

Reply:
@elezar no worries, I think it's fine like this and we can work on the improvements through #102